src/main/java/de/tum/in/www1/artemis/domain/ProgrammingExerciseTestCase.java

```diff
@@ -234,4 +235,29 @@ public String toString() {
         return "ProgrammingExerciseTestCase{" + "id=" + getId() + ", testName='" + testName + '\'' + ", weight=" + weight + ", active=" + active + ", visibility=" + visibility
                 + ", bonusMultiplier=" + bonusMultiplier + ", bonusPoints=" + bonusPoints + '}';
     }
+
+    /**
+     * Check if the provided test was found in the result's feedbacks with positive = true.
+     *
+     * @param result of the build run.
+     * @return true if there is a positive feedback for a given test.
+     */
+    public boolean isSuccessful(Result result) {
+        return result.getFeedbacks().stream().anyMatch(feedback -> {
+            boolean testNameAreSame = feedback.getText() != null && feedback.getText().equalsIgnoreCase(this.getTestName());
```

Comment:
```suggestion
            boolean testNameAreSame = this.getTestName().equalsIgnoreCase(feedback.getText());
```
Or is it possible that this.getTestName() returns null?